Wyvarn/calamus
# Editor Publish a blog

## Description

Allow an authenticated authorized user with editor role to publish a blog

## How can we test this?

1. Run unit tests with `yarn test`
2. Run `docker-compose up`
3. Ensure you have created a user with editor role, by making a POST request like below:

```bash
curl --request POST \
  --url http://localhost:3000/api/v1/register \
  --header 'content-type: application/json' \
  --data '{
	"name" : <NAME>,
	"email": <EMAIL>,
	"password": <PASSWORD>,
	"profilePicUrl": <PIC_URL>,
	"role": "editor"
}'
```
> Any other REST client can be used. Replace all <VALUE> values with actual values

4. Create a blog by making a POST request to create a blog like below:

``` bash
curl --request POST \
  --url http://localhost:3000/api/v1/writer/blog \
  --header 'authorization: Bearer <ACCESS_TOKEN>' \
  --header 'content-type: application/json' \
  --data '{
	"title": <TITLE>,
	"description": <DESCRIPTION>,
	"text": <TEXT>,
	"slug": <SLUG>,
	"imgUrl": <IMAGE_URL>,
	"tags": <ARRAY_OF_TAGS>
}'
```
> Remember to get the <ACCESS_TOKEN> from the registration endpoint from step 3. Replace all <VALUES> with actual values

This will return the blog's id which will be used in the subsequent request.

5. Make a PATCH request to publish a blog:

```bash
curl --request PATCH \
  --url http://localhost:3000/api/v1/editor/blog/publish/5eb317603ab46d5bd36e0d29 \
  --header 'authorization: Bearer <ACCESS_TOKEN>'
```
> The id used here is a sample id

The response returned should be

```json
{
	"statusCode": "10000",
	"message": "Blog published successfully"
}
```

## Checklist:
